rustls/src/crypto/mod.rs

```diff
@@ -227,6 +227,9 @@ pub struct CryptoProvider {
 
     /// Provider for loading private [`SigningKey`]s from [`PrivateKeyDer`].
     pub key_provider: &'static dyn KeyProvider,
+
+    /// Provider for creating [`TicketProducer`]s for stateless session resumption.
+    pub ticketer_factory: Option<&'static dyn TicketerFactory>,
```

Comment:
Thoughts on making this mandatory? A genuine lack of ability to make a ticketer could be expressed by returning an error from `ticketer()`?